components/logins/src/sync/engine.rs

```diff
@@ -343,6 +344,7 @@ impl LoginsSyncEngine {
             self.scope.err_if_interrupted()?;
             Ok(())
         })?;
+        let db = self.store.lock_db()?;
```

Comment:
I think this is causing the deadlocks.  The DB is already locked and you should be able to just reuse the `db` variable.